cliquet/storage/redis.py

```diff
@@ -96,11 +98,32 @@ def _bump_timestamp(self, collection_id, parent_id):
                     pipe.watch(key)
                     previous = pipe.get(key)
                     pipe.multi()
-                    current = utils.msec_time()
+                    is_specified = (record is not None
+                                    and modified_field in record
+                                    or last_modified is not None)
+                    if is_specified:
+                        # If there is a timestamp in the new record,
+                        # try to use it.
+                        if last_modified is not None:
+                            current = last_modified
+                        else:
+                            current = record[modified_field]
+                    else:
+                        current = utils.msec_time()
 
                     if previous and int(previous) >= current:
-                        current = int(previous) + 1
-                    pipe.set(key, current)
+                        collection_timestamp = int(previous) + 1
+                    else:
+                        collection_timestamp = current
+
+                    # Return the newly generated timestamp as the current one
+                    # only if nothing else was specified.
+                    if not is_specified:
+                        current = collection_timestamp
```

Comment:
Since `Redis` is a `MemoryBasedStorage` I believe this piece of code can be shared between the two